include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -469,6 +469,17 @@ class FEEvaluationBase
   value_type
   get_laplacian(const unsigned int q_point) const;
 
+  /**
+   * Return the normal Hessian (i.e., the second derivative of the shape
+   * functions in the direction of the normal vector) of the finite element
+   * function interpolated to the quadrature point with index
+   * @p q_point after a call to @p evaluate(EvaluationFlags::hessians).
+   * Compared to the case when computing the full Hessian, some operations can
+   * be saved when only the normal Hessian is requested.
+   */
+  value_type
+  get_normal_hessian(const unsigned int q_point) const;
+
```

Comment:
I guess we cannot reach the full savings with this interface, as for Cartesian meshes, we could skip most of the computations in sum factorization as well (but we do not know in `evaluate()` or `integrate()` what will be requested at the quadrature points, and we do not know the shape of the element). I think it still makes sense to state the potential savings (as we do for the various specializations of the Hessian like `get_laplacian()`), but we should be aware that a different interface might give us even more benefits.

Reply:
I also have thought about this, and an option would be to introduce a new EvaluationFlag normal_hessians and only do the relevant computations then (with proper assertions if a non-cartesian element is detected). Or also an entirely new function evaluate/integrate normal derivatives.

Reply:
Which is something we can do in the future, these are good ideas.